lib/file.c

```diff
@@ -187,18 +186,31 @@ int file_write_str(char *path, char* buf, int len)
 
 char* file_read_contents(const char *path)
 {
+  int fd;
+  struct stat st;
   char *buf;
-  off_t len;
-  int fd = file_open(path, O_RDONLY);
+
+  /* NOTE: lseek(.. SEEK_END) is broken for this special file type on the
+   * bubblegum board, so we have to use stat instead.
+   */
```

Comment:
if this method works then lets just use this, drop the vendor board specific hack notes as it's not relevant to a mainline geared library where this should be working.